modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/BasePhenotypeAnnotation.pm

```diff
@@ -0,0 +1,970 @@
+
+      if (defined $phenotype->{"study"}) {
+        $st_check_sth->bind_param($param_num,$phenotype->{"study"},SQL_VARCHAR);
+        $param_num++;
+      }
+      if (defined $phenotype->{"study_type"}) {
+        $st_check_sth->bind_param($param_num,$phenotype->{"study_type"},SQL_VARCHAR);
+        $param_num++;
+      }
+      $st_check_sth->execute();
+      $st_check_sth->bind_columns(\$study_id);
+      $st_check_sth->fetch();
+
+      if (!defined($study_id)) {
+        if (length($phenotype->{"study"}) > 255) {
+          print $errFH "WARNING study external_references truncated FROM:>",$phenotype->{"study"}, "<\n";
+          $phenotype->{"study"} = substr($phenotype->{"study"}, 0, 254);
+          $phenotype->{"study"} = substr($phenotype->{"study"}, 0,rindex($phenotype->{"study"}, ",PMID"));
+          print $errFH "WARNING study external_references truncated TO  :>",$phenotype->{"study"}, "<\n";
+        }
+        $st_ins_sth->bind_param(1,$phenotype->{"study"},SQL_VARCHAR);
+        $st_ins_sth->bind_param(2,$phenotype->{"study_type"},SQL_VARCHAR);
+        $st_ins_sth->bind_param(3,$phenotype->{"study_description"},SQL_VARCHAR);
+        $st_ins_sth->execute();
+        $study_id = $db_adaptor->dbc->db_handle->{'mysql_insertid'};
+        $study_count++;
+      }
+    }
+
+    # Remove special characters from the phenotype description
+    foreach my $char (keys(%special_characters)) {
+      my $new_char = $special_characters{$char};
+      $phenotype->{description} =~ s/$char/$new_char/g;
+    }
+
+    # get phenotype ID
+    my $phenotype_id = get_phenotype_id($phenotype, $db_adaptor);
+
+    foreach my $acc (@{$phenotype->{accessions}}){
+      $acc =~ s/\s+//g;
+      $acc = iri2acc($acc) if $acc =~ /^http/;
+      $ontology_accession_ins_sth->execute( $phenotype_id, $acc,  $ont_attrib_type->{$mapped_by}->{'attrib_id'}, $phenotype->{ontology_mapping_type} ) ||die "Failed to import phenotype accession\n";
+    }
+
+    if ($phenotype->{"associated_gene"}) {
+      $phenotype->{"associated_gene"} =~ s/\s//g;
+      $phenotype->{"associated_gene"} =~ s/;/,/g;
+    }
+
+    $phenotype->{"p_value"} = convert_p_value($phenotype->{"p_value"}) if (defined($phenotype->{"p_value"}));
+
+    # Check if this phenotype_feature already exists for this variation and source, in that case we probably want to skip it
+    my $pf_id;
+
+    $pf_check_sth->bind_param(1,$object_id,SQL_VARCHAR);
+    $pf_check_sth->bind_param(2,$object_type,SQL_VARCHAR);
+    $pf_check_sth->bind_param(3,$phenotype_id,SQL_INTEGER);
+    $pf_check_sth->bind_param(4,$source_id,SQL_INTEGER);
+    $pf_check_sth->bind_param(5,$study_id,SQL_INTEGER);
+
+    # For nhgri-ebi gwas data
+    if ($source_info->{source_name_short} =~ m/GWAS/i) {
+      $pf_check_sth->bind_param(6,$phenotype->{"p_value"},SQL_VARCHAR);
+    }
+
+    $pf_check_sth->execute();
+    $pf_check_sth->bind_columns(\$pf_id);
+    $pf_check_sth->fetch();
+    next if (defined($pf_id));
```

Comment:
What is being skipped here?

Reply:
Have added a log print. If there is already a phenotype_feature entry (same object_id, object_type, phenotype_id, source_id, study_id) it won't be inserted again.